### PR TITLE
Compatibility with Bitcoin Core v0.21 

### PR DESCRIPTION
**Pull request by shesek**
As per the changes described in https://bitcoincore.org/en/releases/0.21.0.

v0.21 no longer creates a wallet by default, so we have to create one. I changed the tests to always create a new `testwallet` wallet, let me know if you prefer to only do this for v0.21+.

### CODE REVIEW

**Review by sgeisler — APPROVED** [review]
I didn't check the PR against the API docs diff, but if the tests run with 0.21.0, but I don't see anything obviously wrong with this PR. I might take a closer look if there is no sufficient second review.

ACK ba256d76e343226325645811d79c0a413bbae731

File: json/src/lib.rs
Comment (stevenroose) [inline json/src/lib.rs:562]: nit: I usually put definitions before their usages, but not gonna hold this up for that :)

**Review by stevenroose — APPROVED** [review]
Cool, that `generate` error regression is quite annoying. But this LGTM!

**Pull request merged**
